### PR TITLE
docs: add `CONTRIBUTING.md`

### LINKED ISSUE #1051: Add `CONTRIBUTING.md`
### Please describe the problem you're trying to solve

`CONTRIBUTING.md` can help potential contributors better understand the project structure, required tools, and the process of raising PRs.

### Would you be willing to contribute this feature?

- [ ] Yes, I'll give it a shot

### Describe the solution you'd like

Perhaps we can refer to Rust's `CONTRIBUTING.md` at https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md.

Anyone want to give it a try?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by johan-naizu**
Fixes #1051

### CODE REVIEW

**Comment by sxyazi** [comment]
Thank you, please also add this information:

- Yazi currently requires the nightly version of `rustfmt` and it can be called by `rustfmt +nightly **/*.rs` for formatting if the Rust environment is managed by `rustup`.
- Yazi's documentation repository is located at https://github.com/yazi-rs/yazi-rs.github.io. Contributions related to documentation need to be made within this repository.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+├── yazi-shared/        # Shared utilities
+├── .github/            # GitHub-specific files and workflows
+├── Cargo.toml          # Rust project configuration
+└── README.md           # Project overview
+```
+## Development Setup
+1. Ensure Rust is installed:
+   ```sh
+   rustc --version
+   cargo --version
```
Comment (sxyazi) [inline CONTRIBUTING.md:55]: End tag is missing here

**Comment by johan-naizu** [comment]
I have made the changes

**Review by sxyazi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**